hs_core/serialization.py

```diff
@@ -0,0 +1,1061 @@
+    comments and suggestions on this approach.
+    """
+    def __init__(self):
+        self.root_uri = None
+        # From resource map
+        self.id = None
+        self.res_type = None
+        self.title = None
+        self.files = []
+        # From resource metadata
+        self.abstract = None
+        self.keywords = []
+        self.creators = []
+        self.contributors = []
+        self.coverages = []
+        self.relations = []
+        self.sources = []
+        self.language = None
+        self.rights = None
+        self.creation_date = None
+        self.modification_date = None
+
+        self.bag_content_path = None
+        self.root_uri = None
+        self.res_meta_path = None
+        self._rmeta_graph = None
+
+    @staticmethod
+    def include_resource_file(resource_filename):
+        """
+        :param resource_filename: Name of resource filename.
+        :return: True if resource_filename should be included.
+        """
+        return True
+
+    @classmethod
+    def read_metadata_from_resource_bag(cls, bag_content_path):
+        """
+        Factory method for getting resource-specific metadata from an exploded BagIt archive.
+        To work with this factory, each HydroShare resource type must implement
+        a class that extends GenericResourceMeta stored in the following package/module:
+        ${RESOURCE_PACKAGE}.serialization.${RESOURCE_NAME}Meta.  For example,
+        the "RasterResource", with package "hs_geo_raster_resource" must implement its
+        subclass as hs_geo_raster_resource.serialization.RasterResourceMeta.
+
+        :param bag_content_path: String representing path of exploded bag
+        content.
+
+        :return: An instance of GenericResourceMeta specific to the resource type with all
+        metadata read from the exploded bag.
+
+        :raises: ResourceMetaException if metadata cannot be found or do not appear
+         as expected.
+        """
+
+        # Read resource map so that we know the resource type
+        (root_uri, res_meta_path, res_meta) = cls._read_resource_map(bag_content_path)
+
+        # Iterate over HydroShare resource types
+        res_types = get_resource_types()
+        for rt in res_types:
+            rt_name = rt.__name__
+            if rt_name == res_meta['type']:
+                # Instantiate metadata class for resource type
+                rt_root = rt.__module__.split('.')[0]
+                rt_meta = "{0}Meta".format(rt_name)
+                print("rt_meta: {0}".format(rt_meta))
+                mod_ser_name = "{root}.serialization".format(root=rt_root)
+                print("mod_ser_name: {0}".format(mod_ser_name))
+                instance = None
+                try:
+                    # Use __import__ to make sure mod_ser is compiled on import (else we can't import it)
+                    mod_ser = __import__(mod_ser_name, globals(), locals(), [rt_meta])
```

Comment:
Mostly its just a bit easier to read, and its actually intended for use where as `__import__` is just a mechanism of the import statement and in internal of Python, as such less stable and can change in new versions of python where the importlib utilities have defined stable APIs.